test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,659 @@
+     * with the final incremental directory.
+     */
+    if (counter == 0)
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "%s -R -h %s dump %s > %s.%d", WT_UTIL, home,
+          t_uri, full_out, counter));
+    else
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "%s -R -h %s.%d dump %s > %s.%d", WT_UTIL,
+          home_full, counter, t_uri, full_out, counter));
+    testutil_check(system(buf));
+    /*
+     * Now run dump on the incremental directory.
+     */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s -R -h %s.%d dump %s > %s.%d", WT_UTIL,
+      home_incr, counter, t_uri, incr_out, counter));
+    testutil_check(system(buf));
+
+    /*
+     * Compare the files.
+     */
+    testutil_check(
+      __wt_snprintf(buf, sizeof(buf), "cmp %s.%d %s.%d", full_out, counter, incr_out, counter));
+    ret = system(buf);
+    if (counter == 0)
+        testutil_check(__wt_snprintf(msg, sizeof(msg), "%s", "MAIN"));
+    else
+        testutil_check(__wt_snprintf(msg, sizeof(msg), "%d", counter));
+    printf("Iteration %s: Tables %s.%d and %s.%d %s\n", msg, full_out, counter, incr_out, counter,
+      ret == 0 ? "identical" : "differ");
+    if (ret != 0)
+        exit(1);
+
+    /*
+     * If they compare successfully, clean up.
+     */
+    if (counter != 0) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d %s.%d %s.%d %s.%d", home_full,
+          counter, home_incr, counter, full_out, counter, incr_out, counter));
+        testutil_check(system(buf));
+    }
+    return (ret);
+}
+
+/*
+ * Set up all the directories needed for the test. We have a full backup directory for each
+ * iteration and an incremental backup for each iteration. That way we can compare the full and
+ * incremental each time through.
+ */
+static void
+setup_directories(void)
+{
+    int i;
+    char buf[1024];
+
+    for (i = 0; i <= MAX_ITERATIONS; i++) {
+        /*
+         * For incremental backups we need 0-N. The 0 incremental directory will compare with the
+         * original at the end.
+         */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d && mkdir -p %s.%d/%s",
+          home_incr, i, home_incr, i, logpath));
+        testutil_check(system(buf));
+        if (i == 0)
+            continue;
+        /*
+         * For full backups we need 1-N.
+         */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s.%d && mkdir -p %s.%d/%s",
+          home_full, i, home_full, i, logpath));
+        testutil_check(system(buf));
+    }
+}
+
+static void
+add_work(WT_SESSION *session, const char *t_uri, bool bulk_load)
+{
+    WT_CURSOR *cursor;
+    int i, key;
+    char v[64];
+    /*
+     * Open the cursor with bulk option if bulk_load is true.
+     */
+    if (bulk_load) {
+        testutil_check(session->open_cursor(session, t_uri, NULL, "bulk", &cursor));
+    } else {
+        testutil_check(session->open_cursor(session, t_uri, NULL, NULL, &cursor));
```

Comment:
Instead of passing in a bool you could just pass in the string of NULL or "bulk" and then avoid the conditional.